docs/perpetual-typescript.md

```diff
@@ -0,0 +1,145 @@
+# TypeScript Client
+
+<br>
+<div style="display:flex;">
+  <a href='https://github.com/dydxprotocol/perpetual' style="text-decoration:none;">
+    <img src='https://img.shields.io/badge/GitHub-dydxprotocol%2Fperpetual-lightgrey' alt='GitHub'/>
+  </a>
+  <br>
+  <a href='https://www.npmjs.com/package/@dydxprotocol/perpetual' style="text-decoration:none;padding-left:5px;">
+    <img src='https://img.shields.io/npm/v/@dydxprotocol/perpetual.svg' alt='NPM Package'/>
+  </a>
+</div>
+
+TypeScript library for interacting with the dYdX smart contracts and http API
+
+### Install
+
+```
+npm i -s @dydxprotocol/perpetual
+```
+
+### Initialize
+
+You will need to initialize Perpetual using a [Web3 provider](https://web3js.readthedocs.io/en/v1.2.1/web3.html#providers) / Ethereum node endpoint and Network.
+
+```javascript
+import { Perpetual, Networks } from '@dydxprotocol/perpetual';
+// --- Initialize with Web3 provider ---
+const perpetual = new Perpetual(
+  provider,  // Valid web3 provider
+  Networks.MAINNET,
+  {
+    defaultAccount: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1', // Optional
+    accounts: [
+      {
+        address: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1', // Optional
+        privateKey: '0x4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d',
+      },
+    ], // Optional: loading in an account for signing transactions
+  }, // Optional
+);
+// --- OR Initialize with Ethereum node endpoint ---
+const perpetual = new Perpetual(
+  'https://mainnet.infura.io/v3/YOUR-PROJECT-ID',
+  Networks.MAINNET,
+  {
+    defaultAccount: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1', // Optional - but needed if using Infura
+    accounts: [
+      {
+        address: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1', // Optional
+        privateKey: '0x4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d',
+      },
+    ], // Optional: loading in an account for signing transactions
+  }, // Optional
+);
```

Comment:
removed this option in perpetual. Either way they should pass in a valid provider object